src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,246 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+
+
+/**
+ * A utility class for parsing input and storage.
+ */
+public class Parser {
+    /**
+     * Parses input and returns the corresponding command.
+     *
+     * @param input The input message.
+     * @return The command corresponding to the input message.
+     * @throws BobException If the input is invalid.
+     */
+    public static Command parseInput(String input) throws BobException {
+        if (Objects.equals(input, "")) {
+            return Command.NOOP;
+        }
+
+        String response;
+        String command = input.split(" ")[0];
+        return switch (command) {
+        case "bye" -> new ExitCommand();
+        case "list" -> new ListCommand();
+        case "mark" -> mark(input);
+        case "unmark" -> unmark(input);
+        case "todo" -> todo(input);
+        case "deadline" -> deadline(input);
+        case "event" -> event(input);
+        case "delete" -> delete(input);
+        case "find" -> find(input);
+        default -> throw new BobException("I'm sorry, I did not understand your message.");
+        };
+    }
+
+    /**
+     * Parses a mark message and returns the corresponding mark command.
+     *
+     * @param input The mark message.
+     * @return The mark command corresponding to the mark message.
+     * @throws BobException If the input is invalid.
+     */
+    private static Command mark(String input) throws BobException {
+        Matcher matcher = Pattern.compile("^mark (\\d*)$").matcher(input);
+
+        if (!matcher.find()) {
+            throw new BobException("Please specify which task to mark.");
+        }
+
+        int itemNum = Integer.parseInt(matcher.group(1));
```

Comment:
is itemNum abit too ambiguous? might be better to name it something like itemIndex.